modules/porous_flow/doc/content/modules/porous_flow/tests/infiltration_and_drainage/infiltration_and_drainage_tests.md

```diff
@@ -0,0 +1,276 @@
+viscosity, their effective saturation reads
+\begin{equation}
+S_{\mathrm{eff}} = \frac{1}{\sqrt{1 + \exp((P_{c} - A)/B)}} \ ,
+\label{eqn.rsc.seff}
+\end{equation}
+where $P_{c} = P_{\mathrm{oil}}-P_{\mathrm{water}}$ is the capillary
+pressure, and $A$ and $B$ are arbitrary parameters to be defined by
+the user in the PorousFlow implementation.  For other oil/water viscosity
+ratios $P_{c} = P_{c}(S_{\mathrm{eff}})$ is more complicated, and note
+that their formulation allows $P_{c}<0$, but only
+the particular form [eqn.rsc.seff] need be used to validate
+the MOOSE implementation.
+
+RSC's solutions are quite lengthy, so I will not write them here.  To
+compare with MOOSE, the following parameters are used:
+
+!table id=rsc_params caption=Parameter values used in the Rogers-Stallybrass-Clements infiltration tests
+| Parameter | Value |
+| --- | --- |
+| Bar length | 10 m |
+| Bar porosity | 0.25 |
+| Bar permeability | $10^{-5}\,$m$^{2}$ |
+| Gravity | 0 m.s$^{-2}$ |
+| Water density | 10 kg.m$^{-3}$ |
+| Water viscosity | $10^{-3}\,$Pa.s |
+| Oil density | 20 kg.m$^{-3}$ |
+| Oil viscosity | $2\times 10^{-3}\,$Pa.s |
+| Capillary $A$ | 10 Pa |
+| Capillary $B$ | 1 Pa |
+| Initial water pressure | 0 Pa |
+| Initial oil pressure | 15 Pa |
+| Initial water saturation | 0.08181 |
+| Initial oil saturation | 0.91819 |
+| Water injection rate | 1 kg.s$^{-1}$.m$^{-2}$ |
+
+The input file:
+
+!listing modules/porous_flow/test/tests/infiltration_and_drainage/rsc01.i
+
+In the RSC theory water is injected into a semi-infinite domain,
+whereas of course the MOOSE implementation has finite extent ($0\leq z
+\leq 10$ is chosen).  Because of the near incompressibility of the
+fluids (I choose the bulk modulus to be 2 GPa) this causes the
+porepressures to rise enormously, and the problem can suffer from
+precision-loss problems.  Therefore, the porepressures are fixed at
+$z=10$.  This does not affect the progress of the water saturation
+front.
+
+[rsc.fig] shows good agreement between the analytic
+solution and the MOOSE implementation.  Any minor discrepancies get
+smaller as the temporal and spatial resolution increase, as is
+suggested by the two comparisons in that figure.
+
+The "low-resolution" test has 200 elements in $0\leq z\leq
+10$ and uses 15 time steps is part the automatic test suite that
+is run every time the code is updated.  The "high-resolution" test
+has 600 elements and uses 190 time steps, and is marked as
+"heavy".
+
+!media infiltration_and_drainage/rsc.png style=width:50%;margin-left:10px caption=Water saturation profile after 5 seconds of injection in the Rogers-Stallybrass-Clements test.  The initial water saturation is 0.08181, and water is injected at the top of this figure at a constant rate.  This forms a water front which displaces the oil.  Black line: RSC's analytic solution.  Red squares: high-resolution MOOSE simulation.  Green triangles: lower resolution MOOSE simulation.  id=rsc.fig
```

Comment:
Maybe add a line 

!bibtex bibliography

at the end of this (and the others) to render a "References" title above the references?